### PR TITLE
API-27881-doc-search-flipper-feature

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

Removes 'faraday' flipper features and adds a feature for LH docs service

## Related issue(s)
- [API-27881](https://jira.devops.va.gov/browse/API-27881)

## Testing done

- rspec
- Postman

## What areas of the site does it impact?
	modified:   config/features.yml
	modified:   modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb
	modified:   modules/claims_api/spec/models/veteran/service/user_spec.rb
	modified:   modules/claims_api/spec/requests/v1/power_of_attorney_request_spec.rb
	modified:   modules/claims_api/spec/requests/v1/rswag_power_of_attorney_request_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/power_of_attorney_request_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/rswag_power_of_attorney_spec.rb
	modified:   modules/veteran/app/models/veteran/user.rb
	modified:   modules/veteran/spec/models/veteran/user_spec.rb

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by FonzMP** [comment]
we'll need to make sure we're updating the logic that used these flipper features, specifically the ones we're removing. these would revert back to the old bgs-ext call. 

https://github.com/department-of-veterans-affairs/vets-api/blob/01b5ed22b281616134c348c52a5d4b470acaaed1/modules/veteran/app/models/veteran/user.rb#L29
https://github.com/department-of-veterans-affairs/vets-api/blob/01b5ed22b281616134c348c52a5d4b470acaaed1/modules/veteran/app/models/veteran/user.rb#L39

File: modules/veteran/app/models/veteran/user.rb [hunk 1]
```diff
@@ -26,21 +26,13 @@ def current_poa_code
     end
 
     def current_poa_information
-      @current_poa_information ||= if Flipper.enabled? :bgs_via_faraday
```
Comment (FonzMP) [inline modules/veteran/app/models/veteran/user.rb:29]: I'd add back `@current_poa_information ||=`

**Review by ryan-mcneil — DISMISSED** [review]
👍

File: modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb [hunk 2]
```diff
@@ -237,16 +237,11 @@ def build_representative_info(poa_code)
...
-            ClaimsApi::Logger.log('poa', detail: 'bgs-ext used to locate file_number')
-            bgs_service.people.find_by_ssn(ssn)
-          end
+          ClaimsApi::Logger.log('poa', detail: 'local BGS service used to locate file_number')
+          ClaimsApi::LocalBGS.new(
+            external_uid: target_veteran.participant_id,
+            external_key: target_veteran.participant_id
+          ).find_by_ssn(ssn)
           # rubocop:enable Rails/DynamicFindBy
         end
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:246]: I would say we remove select service and the logging as we know we're using local_bgs now and just pass this straight to the call in `check_file_number_exists!`

File: modules/claims_api/spec/requests/v2/veterans/power_of_attorney_request_spec.rb [hunk 3]
```diff
@@ -15,21 +15,7 @@
...
-    end
-  end
-
-  let(:ows) do
-    if Flipper.enabled? :bgs_via_faraday
-      ClaimsApi::LocalBGS
-    else
-      BGS::OrgWebService
-    end
-  end
```
Comment (FonzMP) [inline modules/claims_api/spec/requests/v2/veterans/power_of_attorney_request_spec.rb:32]: i was confused at first, but now I see. 👍

File: modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb [hunk 4]
```diff
@@ -235,26 +235,20 @@ def build_representative_info(poa_code)
           end
         end
 
-        def select_service(ssn)
+        def local_bgs_service(ssn)
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:238]: I would change this name. if it's specific to `find_by_ssn` now.

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**